controllers/common/ansible.go

```diff
@@ -44,10 +45,19 @@ func CreateAnsibleJob(policyAutomation *policyv1beta1.PolicyAutomation,
 		ansibleJob.Object["spec"].(map[string]interface{})["extra_vars"] = mapExtraVars
 	}
 
-	if targetClusters != nil {
-		// nolint: forcetypeassert
+	if violationContext.TargetClusters != nil {
 		extravars := ansibleJob.Object["spec"].(map[string]interface{})["extra_vars"].(map[string]interface{})
-		extravars["target_clusters"] = targetClusters
+		values := reflect.ValueOf(violationContext)
+		typesOf := values.Type()
+		// add every violationContext fields into extravars
+		// here use the snake key to match the design and
+		// don't need to set target_clusters specifically
+		for i := 0; i < values.NumField(); i++ {
```

Comment:
Optional: Could you perhaps define a new tag (e.g. `ansiblejob`) on each struct field and use that rather than try to automatically convert? Just an idea if we want to be explicit.